src/main/java/uk/ac/ebi/eva/pipeline/io/mappers/AnnotationLineMapper.java

```diff
@@ -101,16 +97,38 @@ public Annotation mapLine(String line, int lineNumber) {
         } else {
             consequenceType.setSoAccessions(mapSoTermsToSoAccessions(lineFields[6].split(",")));
         }
-        currentAnnotation.addConsequenceType(consequenceType);
-        
-        return currentAnnotation;
+        AnnotationMongo currentAnnotation = new AnnotationMongo(
+            AnnotationMongo.buildAnnotationId(
+                variantMap.get("chromosome"),
+                Integer.valueOf(variantMap.get("start")),
+                variantMap.get("reference"),
+                variantMap.get("alternative"),
+                vepVersion,
+                vepCacheVersion),
+            variantMap.get("chromosome"),
+            Integer.valueOf(variantMap.get("start")),
+            Integer.valueOf(variantMap.get("end")),
+            vepVersion,
+            vepCacheVersion,
+            null,
+            Collections.singleton(consequenceType));
+
+        AnnotationMongo AnnotationWithXref = new AnnotationMongo(
```

Comment:
I am not sure what this additional annotation is about??

Reply:
This additional annotation object is because of the way AnnotationMongo.java's class methods are defined in VariationCommons. The `addConsequenceType` and `addConsequenceTypes` functions are private. They are made accessible via `concatenate` command, which accesses `addConsequenceTypes`. `addConsequenceTypes` accesses private `addConsequenceType`.

Additional annotation object is created because as of now, we need to call `concatenate` to access `addConsequenceType` of `AnnotationMongo`. And `concatenate` accepts an `Annotation` as parameter. That is also why 1st I create `ConsequenceTypeMongo`, and then pass it into constructor of `Annotation` object. 


Why we need additional annotation object is because when `concatenate` method also updates `XrefMongo` attribute. That's why I named the additional object `AnnotationWithXref` and did not pass `currentAnnotation` directly.

Referring code snippet [AnnotationMongo.java#L207](https://github.com/EBIvariation/variation-commons/blob/fda0d8bbf44f029ab7ef1f756f31964b3bc9e2c5/variation-commons-mongodb/src/main/java/uk/ac/ebi/eva/commons/mongodb/entities/AnnotationMongo.java#L207)
```
    public AnnotationMongo concatenate(AnnotationMongo annotation) {
        AnnotationMongo temp = new AnnotationMongo(this);
        if (annotation.getConsequenceTypes() != null) {
            temp.addConsequenceTypes(annotation.getConsequenceTypes());
        }
        return temp;
    }
```